code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -187,7 +187,7 @@
 			update_tint()
 		if(G.vision_correction)
 			if(disabilities & NEARSIGHT)
-				overlay_fullscreen("nearsighted", /obj/screen/fullscreen/impaired, 1)
+				become_nearsighted()

```

Comment:
This would also be useless because the function checks if disability exists before adding new overlay.